### PR TITLE
Updating titiler base image to bitnami/python:3.9 in Dockerfile and CI

### PR DESCRIPTION
**Pull request by FischerLGLN**
Signed-off-by: Maik Fischer <maik.fischer@lgln.niedersachsen.de>

## Available PR templates

Issue: https://github.com/developmentseed/titiler/issues/519
Added a bitnami hardened Dockerfile to address security issues.
Python 3.10 is used, since I haven't found build problems with Python 3.10.
I cannot test the ci part, but I've built the image manually and could successful deliver images through titiler.

### CODE REVIEW

File: dockerfiles/Dockerfile.gunicorn [hunk 1]
```diff
@@ -1,8 +1,10 @@
-# Dockerfile for running titiler application with gunicorn server
-# Size ~1.4GB
+# Dockerfile for running titiler application with uvicorn server
 ARG PYTHON_VERSION=3.9
 
-FROM tiangolo/uvicorn-gunicorn:python${PYTHON_VERSION}
+FROM bitnami/python:${PYTHON_VERSION}
+RUN apt update && apt upgrade -y \
+  && apt install curl -y \
+  && rm -rf /var/lib/apt/lists/*
```
Comment (vincentsarago) [inline dockerfiles/Dockerfile.gunicorn:7]: @FischerLGLN Tiangolo image has more than just gunicorn 

I guess the easiest way might just to update https://github.com/vincentsarago/containers/tree/main/uvicorn-gunicorn with the bitnami docker images.

Comment (FischerLGLN) [reply dockerfiles/Dockerfile.gunicorn:7]: You're fast. 

Do you mean this file?
https://github.com/vincentsarago/containers/blob/main/uvicorn-gunicorn/python3.9.dockerfile

I think its mainly the start.sh and start-reload.sh files as well as the guvicorn config, but they aren't needed in a kubernetes platform.
Idea:
We just say guvicorn is for dev and normal use and the uvicorn image is for kubernetes.
I would handle in this PR just the Uvicorn Part and change in kubernetes deployment.

Comment (vincentsarago) [reply dockerfiles/Dockerfile.gunicorn:7]: > I would handle in this PR just the Uvicorn Part and change in kubernetes deployment.

sounds good to me

Comment (FischerLGLN) [reply dockerfiles/Dockerfile.gunicorn]: @vincentsarago Gunicorn is semantically untouched and with this PR I get my updated uvicorn :)

Comment (vincentsarago) [reply dockerfiles/Dockerfile.gunicorn]: 🙏

Comment (vincentsarago) [reply dockerfiles/Dockerfile.gunicorn]: FYI: https://github.com/vincentsarago/containers/pull/3

**Comment by FischerLGLN** [comment]
@vincentsarago 
titiler => (version: "1.0.0", path: "deployment/k8s/charts") > Chart version not ok. Needs a version bump!

Oh wow, I will add a version bump. I suggest bumping 1.0.0 to 1.1.0

**Comment by FischerLGLN** [comment]
@emmanuelmathot FYI: Bumped your titiler helm chart as part of this PR, linting is okay.

**Comment by FischerLGLN** [comment]
@vincentsarago Hi, can this PR be merged? Thanks!

File: dockerfiles/Dockerfile.uvicorn [hunk 2]
```diff
@@ -15,7 +15,7 @@ RUN update-ca-certificates
 ENV CURL_CA_BUNDLE /etc/ssl/certs/ca-certificates.crt
 
 COPY src/titiler/ /tmp/titiler/
-RUN pip install /tmp/titiler/core /tmp/titiler/mosaic /tmp/titiler/application["server"] --no-cache-dir --upgrade
+RUN pip install uvicorn /tmp/titiler/core /tmp/titiler/mosaic /tmp/titiler/application["server"] --no-cache-dir --upgrade
```
Comment (vincentsarago) [inline dockerfiles/Dockerfile.uvicorn:18]: uvicorn is provided by `/tmp/titiler/application["server"]`

[the remark this answers is not in the document]
Comment (FischerLGLN) [reply dockerfiles/Dockerfile.uvicorn]: Thanks, I've removed the last unicorn ;)

**Comment by FischerLGLN** [comment]
@vincentsarago Looking forward to see another package for uvicorn created through master commit: https://github.com/developmentseed/titiler/pkgs/container/titiler

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**